test_flatten.py

```diff
@@ -97,6 +97,16 @@ def test_list_and_dict(self):
         actual = flatten(dic)
         self.assertEqual(actual, expected)
 
+    def test_empty_list_and_dict(self):
+        dic = {
+            'a': {},
```

Comment:
Probably obvious but can you add keys with `None` and `""` values that will also be caught correctly by your modification? Can't think of any other cases but if you can it'd be great to add them.